identity-service/src/typed-routes/solana/solanaRelay.ts

```diff
@@ -0,0 +1,139 @@
+  keys: AccountMetaJSON[]
+  programId: string
+  data: number[]
+}
+
+type RelayRequestBody = {
+  instructions: TransactionInstructionJSON[]
+  skipPreflight: boolean
+  feePayerOverride: string
+  signatures: { signature: number[]; publicKey: string }[]
+  retry: boolean
+  recentBlockhash: string
+  lookupTableAddresses: string[]
+}
+
+const isMalformedInstruction = (instr: TransactionInstructionJSON) =>
+  !instr ||
+  !Array.isArray(instr.keys) ||
+  !instr.data ||
+  !instr.keys.every((key) => !!key.pubkey)
+
+const createRouter = () => {
+  const router = express.Router()
+
+  router.get(
+    '/random_fee_payer',
+    handleResponse(async () => {
+      const feePayerAccount: Keypair = getFeePayerKeypair(false)
+      if (!feePayerAccount) {
+        return errorResponseServerError('There is no fee payer.')
+      }
+      return successResponse({ feePayer: feePayerAccount.publicKey.toBase58() })
+    })
+  )
+
+  router.post(
+    '/relay',
+    parameterizedAuthMiddleware({ shouldRespondBadRequest: false }),
+    handleResponse(<RequestHandler<any, any, RelayRequestBody>>(async (req) => {
+      const libs: AudiusLibs = req.app.get('audiusLibs')
+      const logger = req.logger
+      try {
+        const {
+          instructions: instructionsJSON = [],
+          skipPreflight,
+          feePayerOverride,
+          signatures = [],
+          retry = true,
+          recentBlockhash,
+          lookupTableAddresses = []
+        } = req.body
+
+        // Ensure instructions are formed correctly
+        instructionsJSON.forEach((instr, i) => {
+          if (isMalformedInstruction(instr)) {
+            throw new InvalidRelayInstructionError(i, 'Instruction malformed')
+          }
+        })
+
+        // Parse transactions into TransactionInstructions
+        const instructions = instructionsJSON.map((instr) => {
+          const keys = instr.keys.map((key) => ({
+            pubkey: new PublicKey(key.pubkey),
+            isSigner: key.isSigner,
+            isWritable: key.isWritable
+          }))
+          return new TransactionInstruction({
+            keys,
+            programId: new PublicKey(instr.programId),
+            data: Buffer.from(instr.data)
+          })
+        })
+
+        // Check that the instructions are allowed for relay
+        await assertRelayAllowedInstructions(instructions, { user: req.user })
+
+        // Send transaction using transaction handler
+        const transactionHandler = libs.solanaWeb3Manager!.transactionHandler
+        const {
+          res: transactionSignature,
+          error,
+          errorCode
+        } = await transactionHandler.handleTransaction({
+          recentBlockhash,
+          signatures: (signatures || []).map((s) => ({
+            ...s,
+            signature: Buffer.from(s.signature)
+          })),
+          instructions,
+          skipPreflight,
+          // @ts-ignore TODO: This will be fixed when SDK updates
```

Comment:
nit and likely not your code, but is this error message still relevant?

Reply:
It is new from my code. It's a bit of a chicken/egg thing - you saw this too: https://github.com/AudiusProject/audius-protocol/pull/6190/files/f7909464d1deeee39cde0da22236dab66e98c6ca#r1343112418

The issue is that in `package.json` for identity we have some version of libs, and then when testing this locally I was linking libs. So once I moved to typescript, it was complaining about how the types don't match. The function always accepted a string or a PublicKey